src/trie.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding utf-8 -*-
+
+END_OF_WORD = '$'
+
+
+class Node(object):
+    """Node object.  Has a value and a list of nodes that could come next."""
+
+    def __init__(self, value=None):
+        """
+        Each node will contain the letter at that node and a list of
+        node that come next.
+        """
+        self.value = value
+        self.next = []
+
+    def __eq__(self, other):
+        """Allow nodes to compare and check if a letter is in next list."""
+        return self.value == other
+
+    def _insert(self, word):
+        """
+        Will insert a letter into the list of letters that come next.
+        The remaining part of the word to the next node.
+        """
+        try:
+            letter = Node(word[0])
+        except IndexError:
+            if END_OF_WORD not in self.next:
+                self.next.append(Node(END_OF_WORD))
+            return
+        if letter not in self.next:
+            self.next.append(letter)
+        idx = self.next.index(letter)
```

Comment:
This line is also an O(n) operation.

Check out [https://wiki.python.org/moin/TimeComplexity](https://wiki.python.org/moin/TimeComplexity)